File: Data Structures & Algorithms/merge-strings-alternately/submission-1.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res;
        int n = word1.size();
        int n2 = word2.size();
        bool result = true;
        int i = 0;
        int j = 0;

        while(i < n or j < n2){
            if(i < word1.size() && result == true){
                res += word1[i];
                i++;
                result = false;
            }
            if(j < word2.size() && result == false){
                res += word2[j];
                j++;
                result = true;
            }
        
        if(i == word1.size()){
            result = false;
        }
        if(j == word2.size()){
            result = true;
        }

        }
        return res;
    }
};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        l, r = 0, len(s) - 1

        while l < r:
            if s[l] == s[r]:
                l+=1
                r-=1
            elif s[l + 1] == s[r]:
                r-=1
            elif s[r - 1] == s[l]:
                l+=1
            else:
                return False
        return True

                 
        